driver/timer.hpp

```diff
@@ -139,4 +139,129 @@ class Timer2
     std::chrono::time_point<std::chrono::steady_clock> et;
 };
 
+class RNNCombTimeLoger
+{
+public:
+    RNNCombTimeLoger(hipStream_t main_stream, size_t size, int mode)
+        : stream(main_stream), clockMode(static_cast<ClockMode>(mode))
+    {
+        if(clockMode != ClockMode::Disabled)
+        {
+            hostTimePerLaunch.reserve(size);
+
+            startEvent.reserve(size);
+            endEvent.reserve(size);
+            for(auto i = size; i > 0; --i)
+            {
+                startEvent.push_back(miopen::make_hip_event());
+                endEvent.push_back(miopen::make_hip_event());
+            }
+        }
+    }
+
+    void Start()
+    {
+        if(clockMode == ClockMode::Disabled)
+            return;
+
+        auto launchCont = hostTimePerLaunch.size();
+        hipEventRecord(startEvent[launchCont].get(), stream);
+        st = std::chrono::steady_clock::now();
+    }
+    void StopAndPush()
+    {
+        if(clockMode == ClockMode::Disabled)
+            return;
+
+        std::chrono::time_point<std::chrono::steady_clock> end = std::chrono::steady_clock::now();
+        auto launchCont                                        = hostTimePerLaunch.size();
+
+        hipEventRecord(endEvent[launchCont].get(), stream);
+
+        if(clockMode == ClockMode::OldWallClock)
+        {
+            std::chrono::time_point<std::chrono::steady_clock> st2 =
+                std::chrono::steady_clock::now();
+
+            hipEventSynchronize(endEvent[launchCont].get());
+
+            std::chrono::time_point<std::chrono::steady_clock> end2 =
+                std::chrono::steady_clock::now();
+
+            hostTimePerLaunch.push_back(
+                std::chrono::duration_cast<std::chrono::duration<float, std::milli>>(end - st)
+                    .count() +
+                std::chrono::duration_cast<std::chrono::duration<float, std::milli>>(end2 - st2)
+                    .count());
+        }
+        else
+        {
+            if(clockMode == ClockMode::SeparateClocksSynced)
+                hipEventSynchronize(endEvent[launchCont].get());
+
+            hostTimePerLaunch.push_back(
+                std::chrono::duration_cast<std::chrono::duration<float, std::milli>>(end - st)
+                    .count());
+        }
+    }
+
+    void Print() const
+    {
+        auto n_iter = hostTimePerLaunch.size();
+        if(clockMode == ClockMode::Disabled || n_iter == 0)
+            return;
+
+        float gpu_avg  = 0.;
+        float host_avg = 0.;
+        float gpu_time = 0;
+
+        if(clockMode == ClockMode::SeparateClocksNotSynced)
+            hipEventSynchronize(endEvent[n_iter - 1].get());
+
+        for(auto i = 0ll; i < n_iter; ++i)
```

Comment:
Signed-undigned comparison warning?
```suggestion
        for(auto i = 0ull; i < n_iter; ++i)
```